src/core/ext/filters/client_channel/service_config_channel_arg_filter.cc

```diff
@@ -158,17 +156,11 @@ const grpc_channel_filter ServiceConfigChannelArgFilter = {
 
 void RegisterServiceConfigChannelArgFilter(
     CoreConfiguration::Builder* builder) {
-  builder->channel_init()->RegisterStage(
-      GRPC_CLIENT_DIRECT_CHANNEL, GRPC_CHANNEL_INIT_BUILTIN_PRIORITY,
-      [](ChannelStackBuilder* builder) {
-        auto channel_args = builder->channel_args();
-        if (channel_args.WantMinimalStack() ||
-            !channel_args.GetString(GRPC_ARG_SERVICE_CONFIG).has_value()) {
-          return true;
-        }
-        builder->PrependFilter(&ServiceConfigChannelArgFilter);
-        return true;
-      });
+  builder->channel_init()
+      ->RegisterFilter(GRPC_CLIENT_DIRECT_CHANNEL,
+                       &ServiceConfigChannelArgFilter)
+      .ExcludeFromMinimalStack()
+      .IfHasChannelArg(GRPC_ARG_SERVICE_CONFIG);
```

Comment:
I think this is backwards -- we want to register this filter only if this arg is *not* present.

Reply:
Still need to address this comment, which I'm pretty sure would actually cause a bug (and I suspect we don't have test coverage for it).  Quoting the original comment in case the github UI makes it hard to find:

> I think this is backwards -- we want to register this filter only if this arg is *not* present.

Reply:
Doh!  I was misreading the diff.  Sorry for the confusion here!